packages/block-editor/src/components/line-height-control/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { TextControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import {
+	BASE_DEFAULT_VALUE,
+	RESET_VALUE,
+	STEP,
+	useIsLineHeightControlsDisabled,
+	useLineHeightControlState,
+	isLineHeightDefined,
+} from './utils';
+import { LineHeightControlWrapper } from './styles';
+
+export { default as withLineHeight } from './with-line-height';
+
+export default function LineHeightControl( props ) {
```

Comment:
This is  departure from existing block-editor components where we have value and onChange props. 
I guess there's room for magical components like that that consume directly from attributes but I feel that this is not the right PR to introduce these:

 - These make strong assumptions about  the naming of the attributes
 - If we were to introduce the implicitness involved here, I'd personally prefer if it's a dedicated PR with a plan to generlize this to all block editor components instead of an adhoc one.

Reply:
> I guess there's room for magical components like that that consume directly from attributes but I feel that this is not the right PR to introduce these:

@youknowriad Thank you for your feedback!
Open question! When would it be the right time to introduce this pattern?

I think simplifying Control based components would help simplify the system. Other candidates would be existing color and font size attributes.

> If we were to introduce the implicitness involved here, I'd personally prefer if it's a dedicated PR with a plan to generlize this to all block editor components instead of an adhoc one.

I thought it being behind an `__experimental` prefix would have been okay.
I viewed this as more a first attempt at this approach since it would only be scoped to the Paragraph block.

Reply:
> I viewed this as more a first attempt at this approach since it would only be scoped to the Paragraph block.

I think that's a good way to introduce it iteratively. I know @mtias likes this pattern but I have concerns personally though. The main one being the implicit attribute presence. 

One way to lower that issue though would be to use a generic attribute. For example `style` could work as it can be an attribute used for any global style variables. 

If we use separate attributes, I'd worry about the proliferation of adhoc attributes.